### PR TITLE
Add outside_collaborators list

### PR DESCRIPTION
**Pull request by Diegunix**
# Description

This PR adds the ability to list outside members of the organization.
As indicated in the github documentation (https://docs.github.com/en/rest/orgs/outside-collaborators)

# Before submitting a PR:

- [X] Changes must not break binary backwards compatibility. If you are unclear on how to make the change you think is needed while maintaining backward compatibility, [CONTRIBUTING.md](CONTRIBUTING.md) for details.
- [X] Add JavaDocs and other comments as appropriate. Consider including links in comments to relevant documentation on https://docs.github.com/en/rest .  
- [x] Add tests that cover any added or changed code. This generally requires capturing snapshot test data. See [CONTRIBUTING.md](CONTRIBUTING.md) for details.
- [ ] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.
- [X] Push your changes to a branch other than `main`. You will create your PR from that branch.

# When creating a PR: 

- [X] Fill in the "Description" above with clear summary of the changes. This includes:
  - [ ] If this PR fixes one or more issues, include "Fixes #<issue number>" lines for each issue. 
  - [X] Provide links to relevant documentation on https://docs.github.com/en/rest where possible.
- [x] All lines of new code should be covered by tests as reported by code coverage. Any lines that are not covered must have PR comments explaining why they cannot be covered. For example, "Reaching this particular exception is hard and is not a particular common scenario."
- [X] Enable "Allow edits from maintainers".

### CODE REVIEW

**Review by bitwiseman — CHANGES_REQUESTED** [review]
Add tests please.

**Comment by Diegunix** [comment]
> Add tests please.

DONE

File: src/main/java/org/kohsuke/github/GHOrganization.java [hunk 1]
```diff
@@ -315,6 +315,17 @@ public PagedIterable<GHUser> listPublicMembers() throws IOException {
...
 
+    /**
+     * All the outside collaborators of this organization.
+     *
+     * @return the paged iterable
+     * @throws IOException
+     *             the io exception
+     */
+    public PagedIterable<GHUser> listOutsideCollaborators() throws IOException {
+        return listMembers("outside_collaborators");
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHOrganization.java:326]: Please also cover this method with a test.  The default filter is `all`. 
 Adding one more call and some verifications to the test you wrote shouldn't be too much work.  And the list returned from the call should be identical to the call already made in your test.

**Review by bitwiseman — CHANGES_REQUESTED** [review]
A little more testing and I'll merge.  Thanks for contributing!

File: src/main/java/org/kohsuke/github/GHOrganization.java (hunk 1 above)
Comment (Diegunix) [reply src/main/java/org/kohsuke/github/GHOrganization.java:326]: DONE

**Review by bitwiseman — APPROVED** [review]

**Pull request merged**